### PR TITLE
Add support for wildcards, message, defualtFile, DefaultDir and DefatltPath for Choosers

### PR DESCRIPTION
**Pull request by gonzygonz**
#363
adding this feature
i had to delete previous PR as i did some mess there (again, im new to git and github)
i did the changes you suggested to use .get() method
now its a much cleaner implementation.

let me know if you need anything else

P.S after this PR is done, i will find time to update the example to use this new feature

### CODE REVIEW

File: gooey/gui/components/widgets/core/chooser.py [hunk 1]
```diff
@@ -64,40 +64,57 @@ def getValue(self):
...
 class FileSaver(Chooser):
     """ Specify the path to save a new file """
     def getDialog(self):
+        options = self.Parent._options
         return wx.FileDialog(
             self,
             style=wx.FD_SAVE | wx.FD_OVERWRITE_PROMPT,
-            defaultFile=_("enter_filename"),
-            message=_('choose_file')
+            defaultFile=options.get('defaultFile', _("enter_filename")),
```
Comment (chriskiehl) [inline gooey/gui/components/widgets/core/chooser.py]: These option keys should actually be snake_cased as they come from Gooey's public options API, which honors PEP8.

File: gooey/gui/components/widgets/core/chooser.py [hunk 2]
```diff
@@ -64,40 +64,57 @@ def getValue(self):
 class FileChooser(Chooser):
     """ Retrieve an existing file from the system """
     def getDialog(self):
-        return wx.FileDialog(self, _('open_file'), style=wx.FD_OPEN | wx.FD_FILE_MUST_EXIST)
+        options = self.Parent._options
+        return wx.FileDialog(self, message=options.get('message', _('open_file')),
+                             style=wx.FD_OPEN | wx.FD_FILE_MUST_EXIST,
+                             defaultFile=options.get('defaultFile', _("enter_filename")),
```
Comment (chriskiehl) [inline gooey/gui/components/widgets/core/chooser.py]: You'll need to add these new options to the Gooey Options docs here: https://github.com/chriskiehl/Gooey/blob/1.0.3-release/docs/Gooey-Options.md#individual-widget-options

File: gooey/gui/components/widgets/core/chooser.py (hunk 1 above)
Comment (gonzygonz) [reply gooey/gui/components/widgets/core/chooser.py]: hi
i kept options as they appear in original wx lib so users will have the same interface in Gooey as they have in wx.
besides - its a string dict key, so i dont think pep8 has any case enforce on this
(make sure you are not confused by the removed code which had variables and they really should have been snake_case)

i think its smarter to have it the same as wx, but if you feel like it should be snake_case i will change that.

Comment (chriskiehl) [reply gooey/gui/components/widgets/core/chooser.py]: As much as possible, we shouldn't be exposing wx internals to outside clients or using it as a rational for driving APIs. Yes, we happen to be using the same keyname as wx here. Their names are Good Enough that it doesn't matter -- `default_dir` has no coupling to wx specifically -- we could rewrite Gooey with Tkinter or PyQt and use the reuse the name. Additionally, your options would be the _only_ options in Gooey's public API that are camelCased. That alone should be reason enough to snake_case.

**Comment by gonzygonz** [comment]
Hi
i updated to snake_case and added documentation.
first time i do github md documentation so ill be happy for notes (if any)

**Comment by eladeyal-intel** [comment]
@gonzygonz I propose that the docuentation can be clearer about when to use "default_path" and when to use "default_dir" and "default_file".

Also I propose not to use `message` as a parameter name. I think it the term is too broad to be useful. Perhaps `prompt` or `dialog_prompt` is better?

**Comment by eladeyal-intel** [comment]
@gonzygonz `title` should be used in my opinion only for the text which is in the window title bar. If the text in question is beneath the title bar (of if the window is modal and there is no title bar) then I propose `prompt`

File: gooey/gui/components/widgets/core/chooser.py (hunk 2 above)
Comment (gonzygonz) [reply gooey/gui/components/widgets/core/chooser.py]: I've added documentation.
Please review it is clear.

**Comment by chriskiehl** [comment]
Long delay, but looks good! Thanks for the addition!

**Pull request merged**